reference/latest/src/client/java/com/example/docs/network/basic/FabricDocsReferenceNetworkingBasicClient.java

```diff
@@ -0,0 +1,60 @@
+package com.example.docs.network.basic;
+
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LightningEntity;
+import net.minecraft.entity.SpawnReason;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;
+import net.fabricmc.fabric.api.event.player.UseEntityCallback;
+
+import com.example.docs.networking.basic.GiveGlowingEffectC2SPayload;
+import com.example.docs.networking.basic.SummonLightningS2CPayload;
+
+public class FabricDocsReferenceNetworkingBasicClient implements ClientModInitializer {
+	@Override
+	public void onInitializeClient() {
+		// :::client_global_receiver
+		ClientPlayNetworking.registerGlobalReceiver(SummonLightningS2CPayload.ID, (payload, context) -> {
+			ClientWorld world = context.client().world;
+
+			if (world == null) {
+				return;
+			}
+
+			BlockPos lightningPos = payload.pos();
+			LightningEntity entity = EntityType.LIGHTNING_BOLT.create(world, SpawnReason.TRIGGERED);
+
+			if (entity != null) {
+				entity.setPosition(lightningPos.getX(), lightningPos.getY(), lightningPos.getZ());
+				world.addEntity(entity);
+			}
+		});
+		// :::client_global_receiver
+
+		// :::use_entity_callback
+		UseEntityCallback.EVENT.register((player, world, hand, entity, hitResult) -> {
+			if (!world.isClient()) {
+				return ActionResult.PASS;
+			}
+
+			ItemStack usedItemStack = player.getStackInHand(hand);
+
+			if (usedItemStack.isOf(Items.POISONOUS_POTATO) && hand == Hand.MAIN_HAND) {
```

Comment:
The client may send a packet that'll be rejected by our check if we interact with a non-living entity (e.g. armor stand or minecart)
```suggestion
			if (entity instanceof LivingEntity && usedItemStack.isOf(Items.POISONOUS_POTATO) && hand == Hand.MAIN_HAND) {
```

Reply:
Armor stands are living entities, but this would be valid for things like mine carts. So lets keep it.